astropy/table/table.py

```diff
@@ -896,24 +900,30 @@ def show_in_notebook(self, tableid=None, css=None, display_length=50,
                                             np.random.randint(1, 1e6))
 
         jsv = JSViewer(display_length=display_length)
-        html = self._base_repr_(html=True, max_width=-1, tableid=tableid,
-                                max_lines=-1, show_dtype=False,
-                                tableclass=table_class)
+
+        has_row_index = 'row_index' in self.colnames
```

Comment:
I see your point re: `row_index`.  What do you think about `idx`?  That gets across the point that it's a python level index, not a "row" like in a spreadsheet (which is usually 1-indexed).